### PR TITLE
Add setProcessingThreadNames context parameter.

### PR DESCRIPTION
**Pull request by gianm**
Believe it or not, setting thread names takes a measurable amount of time in the case where segment scans are very quick. In high-QPS testing we found a slight performance boost from turning off processing thread renaming. This option makes that possible.

### CODE REVIEW

File: processing/src/main/java/org/apache/druid/query/spec/SpecificSegmentQueryRunner.java [hunk 1]
```diff
@@ -45,6 +46,9 @@
   private final QueryRunner<T> base;
   private final SpecificSegmentSpec specificSpec;
 
+  @VisibleForTesting
+  static final String CTX_SET_THREAD_NAME = "setProcessingThreadNames";
```
Comment (cryptoe) [inline processing/src/main/java/org/apache/druid/query/spec/SpecificSegmentQueryRunner.java:50]: Should we document this somewhere?
Maybe Query context.md ?

Comment (gianm) [reply processing/src/main/java/org/apache/druid/query/spec/SpecificSegmentQueryRunner.java:50]: Added documentation.

File: processing/src/main/java/org/apache/druid/query/spec/SpecificSegmentQueryRunner.java [hunk 2]
```diff
@@ -64,18 +68,26 @@ public Sequence<T> run(final QueryPlus<T> input, final ResponseContext responseC
...
+    final Query<T> query = queryPlus.getQuery();
 
-    final Sequence<T> baseSequence = doNamed(
-        currThread,
-        currThreadName,
-        newName,
-        () -> base.run(queryPlus, responseContext)
-    );
+    final Thread currThread = setName ? Thread.currentThread() : null;
+    final String currThreadName = setName ? currThread.getName() : null;
```
Comment (cryptoe) [inline processing/src/main/java/org/apache/druid/query/spec/SpecificSegmentQueryRunner.java:76]: Super Nit: 75:77 we are checking the setName variable thrice.
We can easily move them inside  the block which starts from line 81. 
I think it makes the code easier to read as well.

**Review by cryptoe — APPROVED** [review]
LGTM!!

File: processing/src/main/java/org/apache/druid/query/spec/SpecificSegmentQueryRunner.java (hunk 2 above)
Comment (gianm) [reply processing/src/main/java/org/apache/druid/query/spec/SpecificSegmentQueryRunner.java:76]: I tried that at first, and then realized there's another usage of these variables at line 164.

**Review by abhishekagarwal87 — APPROVED** [review]

**Pull request merged**